### ISSUE TITLE
type inconsistency between `Collection::size()` and index for const object accessors

### ISSUE THREAD

**Issue by c-dilks** [AIDASoft/podio#406]
- PODIO version: `master`

There is a difference between the type `size_t` returned by `size()` from a Collection and the type of the index used in the const object accessors, `unsigned int`:

https://github.com/AIDASoft/podio/blob/f4c9219ddde59b1efc73db3fd9c25139222a7b3c/python/templates/Collection.h.jinja2#L77

vs.

https://github.com/AIDASoft/podio/blob/f4c9219ddde59b1efc73db3fd9c25139222a7b3c/python/templates/Collection.h.jinja2#L94-L101

Is there a particular reason for this difference?

_cf._ `std::vector`, where we have (ignoring C++20 changes for clarity):
```cpp
size_type size();
reference at( size_type pos );
```
in other words, `std::vector::at` takes the same type that `std::vector::size()` returns.

**Comment by tmadlener** [comment]
You are right. I don't see a reason to not be consistent here. This should be fairly straight forward to fix, I think as these are directly passed to the underlying `std::deque`, e.g.

https://github.com/AIDASoft/podio/blob/f4c9219ddde59b1efc73db3fd9c25139222a7b3c/python/templates/Collection.cc.jinja2#L39-L41

Out of of curiosity: Does something change with c++20 in this respect? I am asking since you mentioned it explicitly.

**Comment by tmadlener** [comment]
It also looks like we have a `size_t` in the `RelationRange` that is one of the returns above.
https://github.com/AIDASoft/podio/blob/f4c9219ddde59b1efc73db3fd9c25139222a7b3c/include/podio/RelationRange.h#L31-L33

For consistency with the usual c++ containers I think it would be good to have `size_t` everywhere. From a purely functional point of view, it doesn't really matter probably, because the `ObjectID` uses an `int` internally at the moment, and that is essentially the limiting factor for the number of elements in a collection:
https://github.com/AIDASoft/podio/blob/f4c9219ddde59b1efc73db3fd9c25139222a7b3c/include/podio/ObjectID.h#L9-L10
